src/java/com/phenix/pct/PCTCompile.java

```diff
@@ -277,13 +279,11 @@ private boolean createDir(File dir) {
      */
     public void execute() throws BuildException {
         // Create dest directory if necessary
-        if (compAttrs.getDestDir() == null) {
-            cleanup();
-            throw new BuildException(Messages.getString("PCTCompile.34")); //$NON-NLS-1$
-        }
-        if (!createDir(compAttrs.getDestDir())) {
-            cleanup();
-            throw new BuildException(MessageFormat.format(Messages.getString("PCTCompile.36"), "destDir")); //$NON-NLS-1$
+        if (compAttrs.getDestDir() != null) {
+            if (!createDir(compAttrs.getDestDir())) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1066)